### PR TITLE
Don't panic on malformed NFS traffic

### PR DESCRIPTION
**Pull request by DominoTree**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2397

Describe changes:
Removed a call to the panic!() macro if an unsupported/invalid NFS version is encountered. Not sure this is the correct way to do this, but it seems to have solved the issue in my environment. Perhaps this might make more sense with a Result<> wrapper or something?

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: rust/src/nfs/nfs.rs [hunk 1]
```diff
@@ -1069,7 +1069,10 @@ impl NFSState {
                 SCLogDebug!("NFSv2 reply record");
                 return self.process_reply_record_v2(r, &xidmap);
             },
-            _ => { panic!("unsupported NFS version"); },
+            _ => {
+                SCLogNotice!("unsupported NFS version");
```
Comment (regit) [inline rust/src/nfs/nfs.rs]: Looks like having a NFSv4 message would be legit so we should ignore this value.

For weird value, using a log notice message could introduce a message storm if ever a lot of weird packets are sent. It is better to use a Debug level message and set an event instead with 
```
self.set_event(NFSEvent::MalformedData);
```
Ideally, a dedicated event like NonExistingVersion would be better.

Comment (DominoTree) [reply rust/src/nfs/nfs.rs]: Changes seem simple enough, I'll get them sorted :)

**Comment by DominoTree** [comment]
Went ahead and added an NFS event for `UnsupportedVersion` as well because it seemed to make sense. If not, let me know and I'll go ahead and remove it.

**Comment by victorjulien** [comment]
Merged in #3133, thanks! I did a small commit msg cleanup.

**Pull request closed without merging**